services/sharedchannel/sync_recv.go

```diff
@@ -306,6 +317,8 @@ func (scs *Service) updateSyncUser(patch *model.UserPatch, user *model.User, cha
 				)
 			}
 		} else {
+			scs.server.GetStore().User().InvalidateProfileCacheForUser(update.New.Id)
```

Comment:
This will not invalidate caches all across the cluster. Should we use `a.InvalidateCacheForUser` instead?

Reply:
Good point!   Fixed via https://github.com/mattermost/mattermost-server/pull/17361/commits/33adb5730da49e0f48f24a4d67a9bf54d9cb435b